lib/c0/c0sk_internal.c

```diff
@@ -691,6 +704,20 @@ c0sk_ingest_worker(struct work_struct *work)
 
     ingest->t6 = get_time_ns();
 
+    /* Initialize the per-thread throttle state so that c0sk_cningest_cb() (called
+     * repeatedly by bkv_collection_finish_pair()) can update the c0sk throttle
+     * sensor with the observed spill rate.
+     */
+    if (c0sk->c0sk_sensor) {
+        ingest->c0iw_thr_sensor = c0sk->c0sk_sensor;
+        ingest->c0iw_thr_tls = &hse_throttle_tls;
```

Comment:
Why not use the thread local var directly instead of storing it in the ingest worker struct?  Storing a TLS ptr in a struct opens up the possibility of using it in another thread (if the struct is used by another thread).

Reply:
hmm, lemme review...

Reply:
It has to be initialized before the call to bkv_collection_finish_pair(), the latter which calls c0sk_cningest_cb() for each and every k/v tuple.  finish-pair doesn't have any visibility into c0sk, it just walks the binheap and calls the callback. 

But more importantly, saving the tls ptr here eliminates a call to __tls_getaddr() for every k/v tuple as that function isn't particularly cheap (although it's cheaper now since linux 5.something, https://www.kernel.org/doc/html/latest/x86/x86_64/fsgs.html).

I doubt it's humanly possible to make bkv_collection_finish_pair() multithreaded, considering it has to sort all the keys from many binheaps, so another thread using it shouldn't be an issue.  I could null out the tls ptr after that call and add a comment to that effect...

Reply:
I nulled out the tls ptr after the call to finish-pair().